crowbar_framework/app/models/deployer_service.rb

```diff
@@ -77,7 +77,8 @@ def transition(inst, name, state)
     if state == "hardware-installing"
       node = NodeObject.find_node_by_name(name)
       # build a list of current and pending roles to check against
-      roles = node.roles.dup
+      roles = []
+      roles = node.roles.dup if node.roles
```

Comment:
Isn't this a classic case for the tenary operator?

```roles =  node.roles ? node.roles.dup : [] ```

I think that would be somewhat cleaner, but if you prefer your code. Feel free to merge.